commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -619,6 +620,64 @@ def update_line_item_transition_state_on_fulfillment(
             handle_commercetools_error(err, f"Unable to update LineItemState of order {order_id}", True)
             return None
 
+    def update_all_line_items_transition_state_on_fulfillment(
+            self,
+            order_id: str,
+            order_version: int,
+            line_items: List[CTLineItem],
+            from_state_id: str,
+            new_state_key: str,
+    ) -> CTOrder:
+        """
+        Update Commercetools order line item state for all items in one call.
+        Args:
+            order_id (str): Order ID (UUID)
+            order_version (int): Current version of order
+            line_items (List[object]): List of line item objects
+            from_state_id (str): ID of LineItemState to transition from
+            new_state_key (str): Key of LineItemState to transition to
+        Returns (CTOrder): Updated order object or
+        Returns (CTOrder): Current un-updated order
+        Raises Exception: Error if update was unsuccessful.
+        """
+
+        from_state_key = self.get_state_by_id(from_state_id).key
+
+        logger.info(
+            f"[CommercetoolsAPIClient] - Transitioning all line item states for order with ID {order_id} "
+            f"from {from_state_key} to {new_state_key}"
+        )
+
+        try:
+            if new_state_key != from_state_key:
+                actions = [
+                    OrderTransitionLineItemStateAction(
+                        line_item_id=item.id,
+                        quantity=item.quantity,
+                        from_state=StateResourceIdentifier(key=from_state_key),
+                        to_state=StateResourceIdentifier(key=new_state_key),
+                    )
+                    for item in line_items
+                ]
+
+                updated_fulfillment_line_item_order = self.base_client.orders.update_by_id(
+                    id=order_id,
+                    version=order_version,
+                    actions=actions,
+                )
+
+                return updated_fulfillment_line_item_order
+            else:
+                logger.info(
```

Comment:
Is there a chance that some of our lineItems got updated but some of them didn't? What would be the flow in that case?